content/post/run-elasticsearch-in-aws/index.md

```diff
@@ -0,0 +1,692 @@
+  - `topology.master` - specifies the properties of master nodes.
+    - `master.replicas` - specifies the number of master nodes.
+    - `master.storage` - specifies the master node storage information that passed to the StatefulSet.
+  - `topology.data` - specifies the properties of data nodes.
+    - `data.replicas` - specifies the number of data nodes.
+    - `data.storage` - specifies the data node storage information that passed to the StatefulSet.
+  - `topology.ingest` - specifies the properties of ingest nodes.
+    - `ingest.replicas` - specifies the number of ingest nodes.
+    - `ingest.storage` - specifies the ingest node storage information that passed to the StatefulSet.
+
+Once these are handled correctly and the Elasticsearch object is deployed, you will see that the following objects are created:
+
+```bash
+$ kubectl get all -n demo
+NAME                               READY   STATUS    RESTARTS   AGE
+pod/es-topology-cluster-data-0     1/1     Running   0          118s
+pod/es-topology-cluster-data-1     1/1     Running   0          79s
+pod/es-topology-cluster-ingest-0   1/1     Running   0          118s
+pod/es-topology-cluster-ingest-1   1/1     Running   0          81s
+pod/es-topology-cluster-master-0   1/1     Running   0          118s
+pod/es-topology-cluster-master-1   1/1     Running   0          74s
+
+NAME                                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+service/es-topology-cluster          ClusterIP   10.100.102.136   <none>        9200/TCP   2m2s
+service/es-topology-cluster-master   ClusterIP   None             <none>        9300/TCP   2m2s
+service/es-topology-cluster-pods     ClusterIP   None             <none>        9200/TCP   2m2s
+
+NAME                                          READY   AGE
+statefulset.apps/es-topology-cluster-data     2/2     2m4s
+statefulset.apps/es-topology-cluster-ingest   2/2     2m4s
+statefulset.apps/es-topology-cluster-master   2/2     2m4s
+
+NAME                                                     TYPE                       VERSION   AGE
+appbinding.appcatalog.appscode.com/es-topology-cluster   kubedb.com/elasticsearch   8.2.0     2m8s
+
+NAME                                           VERSION       STATUS   AGE
+elasticsearch.kubedb.com/es-topology-cluster   xpack-8.2.0   Ready    2m30s
+```
+Let’s check if the database is ready to use,
+
+```bash
+$ kubectl get elasticsearch -n demo es-topology-cluster
+NAME                  VERSION       STATUS   AGE
+es-topology-cluster   xpack-8.2.0   Ready    3m1s
+```
+> We have successfully deployed Elasticsearch in Amazon EKS. Now we can exec into the container to use the database.
+
+### Insert Sample Data
+
+In this section, we are going to create few indexes in Elasticsearch. At first, we are going to port-forward the respective Service so that we can connect with the database from our local machine. Then, we are going to insert some data into the Elasticsearch.
```

Comment:
On the deployment of Elasticsearch yaml, the operator creates a governing service that is named after the Elasticsearch object name itself. We are going to use this object to....